### PR TITLE
Example 2 (Celsius to Fahrenheit) Convert done

### PR DESCRIPTION
**Pull request by F20170604**
Original File Link : https://colab.research.google.com/github/tensorflow/examples/blob/master/courses/udacity_intro_to_tensorflow_for_deep_learning/l02c01_celsius_to_fahrenheit.ipynb#scrollTo=xrpFFlgYhCty

Converted File Link: https://colab.research.google.com/drive/1Gp5ypYsDZ3QPzgNBCUbW3h6fMj3asz9y

### CODE REVIEW

File: S4TF_Tutorial_2.ipynb
Comment (BradLarson) [inline S4TF_Tutorial_2.ipynb]: This second `import TensorFlow` can probably be removed, because it was already imported above.

Comment (F20170604) [reply S4TF_Tutorial_2.ipynb]: Oh yes sure 👍

**Comment by rxwei** [comment]
Could you please review the [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines) and the [Google Swift Style Guide](https://google.github.io/swift) to make sure your code conforms to those guidelines? In particular, if you could spend some time to make sure the name capitalization and horizontal whitespaces follow the guidelines, that will be very helpful!

> Note: The only exception from the Google Swift Style Guide is that we apply 4-space indentation.

**Comment by F20170604** [comment]
> Could you please review the [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines) and the [Google Swift Style Guide](https://google.github.io/swift) to make sure your code conforms to those guidelines? In particular, if you could spend some time to make sure the name capitalization and horizontal whitespaces follow the guidelines, that will be very helpful!
> 
> > Note: The only exception from the Google Swift Style Guide is that we apply 4-space indentation.

Sure

**Comment by F20170604** [comment]
@rxwei I have made the necessary changes. For variables, lowerCamelCase is used and 4-space indentation is used in the entire notebook wherever needed. Can you please verify them?
@dynamicwebpaige Can you please check if the explanation needs any changes for this notebook? :smile:

**Comment by rxwei** [comment]
Thanks Ayush! I’ll take a closer look tomorrow or Tuesday.

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]: Please make sure there’s a whitespace after every colon, comma or other punctuation.

Comment (F20170604) [reply S4TF_Tutorial_2.ipynb]: Sure 👍 
Done

**Comment by dynamicwebpaige** [comment]
Thanks for making these changes, @Ayush517! @rxwei, are there any other modifications that Ayush should make before this is merged?

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]: Type names should start with an uppercase letter.

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]: Remove extra lines after `{` and before `}`.

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]: Type names should start with an uppercase letter. 

Also, what does "Exp" mean? The Swift API Design Guidelines actually discourage abbreviations. How about renaming `c2fConv` to `CelsiusToFahrenheit`, and renaming `c2fConvExp` to `CelsiusToFahrenheit2`?

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]: [`TensorShape` conforms to `ExpressibleByArrayLiteral`](https://github.com/tensorflow/swift-apis/blob/c1b18f73be9d67c5b2be80c5dfd044791c0c1d2d/Sources/TensorFlow/Core/TensorShape.swift#L23). You can use `[7, 1]` directly.

```suggestion
        "let input = Tensor<Float>(shape: [7, 1], scalars: celsiusTemperatures)\n",
```

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]:
```suggestion
        "let output = Tensor<Float>(shape: [7, 1], scalars: fahrenheitTemperatures)\n",
```

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]: Remove extra whitespace.
```suggestion
        "let fahrenheitTemperatures: [Float] = [-40, 14, 32, 46, 59, 72, 100]\n",
```

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]:
```suggestion
        "var model = CelciusToFahrenheit2()\n",
```

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]:
```suggestion
        "let optimizer = Adam(for: model2, learningRate: 0.1)\n",
```

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [reply S4TF_Tutorial_2.ipynb]: New names will shadow old ones of the same name. No need to disambiguate with "exp" for any variables here.

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]:
```suggestion
        "    if epoch.isMultiple(of: 20) {\n",
```

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]: Could you correct the indentation here?
```suggestion
        "    optimizer.update(&model.allDifferentiableVariables, along: grad)\n",
```

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]:
```suggestion
        "    if epoch.isMultiple(of: 20) {\n",
```

File: S4TF_Tutorial_2.ipynb
Comment (F20170604) [reply S4TF_Tutorial_2.ipynb]: This was a part of an experiment for the learners, so I named it as Exp

Comment (rxwei) [reply S4TF_Tutorial_2.ipynb]: Understood. However, since the entire section is titled "A little experiment", adding "exp" to each variable feels not adding much value but just redundancy. Also, because Swift generally doesn't encourage the use of abbreviations and because "exp" has many possible meanings, names ending with "Exp" can be very confusing.

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]:
```suggestion
        "    let (loss2, grad2) = model2.valueWithGradient { model -> Tensor<Float> in\n",
```

Drop redundant type information.

File: S4TF_Tutorial_2.ipynb
Comment (rxwei) [inline S4TF_Tutorial_2.ipynb]:
```suggestion
        "    let (loss, grad) = model.valueWithGradient { model -> Tensor<Float> in\n",
```

**Review by rxwei — COMMENTED** [review]
In the experiment section, I don't think it's particularly meaningful to suffix any variable with `2` (or `Exp` earlier). They are already in a section titled "experiment". So long as the model type is different (`CelsiusToFahrenheit2`), it's less confusing to the reader to just use variable names like `model`, `loss` and `grad`.

**Review by rxwei — APPROVED** [review]
LGTM after comments above are addressed. Thanks!

**Comment by F20170604** [comment]
Can this be merged now? All issues addressed @rxwei

**Comment by dynamicwebpaige** [comment]
@Ayush517 @rxwei: SGTM! Thanks to you both for reviewing and resolving the issues. Excited to have this notebook as an example!

**Comment by rxwei** [comment]
Looks great. Thanks for your patience in addressing PR feedback!

**Comment by F20170604** [comment]
> Looks great. Thanks for your patience in addressing PR feedback!

Oh certainly. I loved it. Learnt so so many things during this.
I hope feedbacks like this happen to all the further PRs. :+1:

**Pull request merged**
